crates/allocator/src/bump.rs

```diff
@@ -174,9 +174,10 @@ impl InnerAlloc {
     /// Note: This implementation results in internal fragmentation when allocating across
     /// pages.
     fn alloc(&mut self, layout: Layout) -> Option<usize> {
-        let alloc_start = self.next;
+        let alloc_start = self.get_aligned_ptr(&layout);
 
         let aligned_size = layout.pad_to_align().size();
```

Comment:
We don't need to align the size as me make sure that the `alloc_start` is always aligned.